source/core_http_client.c

```diff
@@ -24,8 +24,11 @@
  * @file core_http_client.c
  * @brief Implements the user-facing functions in core_http_client.h.
  */
-
-#include <assert.h>
+#ifdef DISABLE_ASSERT
```

Comment:
This is what the "NDEBUG" macro from the c-standard is for.

Reply:
Hadn't realized that this macro existed until this point to be honest. Should I go through the various repositories and just set this macro instead of re-defining the asserts?

Reply:
It shouldn't be necessary at all. According to the [C89 standard](http://port70.net/~nsz/c/c89/c89-draft.html#:~:text=i%20%3D%20atoi(str)%3B-,4.2%20DIAGNOSTICS%20%3Cassert.h%3E,-The%20header%20%3Cassert):


> If NDEBUG is defined as a macro name at the point in the source file where <assert.h> is included, the assert macro is defined simply as
> ``` 
> #define assert(ignore) ((void)0) 
> ```